### PR TITLE
TST: simplify how the different code paths for block are tested.

### PR DESCRIPTION
**Pull request by hmaarrfk**
xref: https://github.com/numpy/numpy/pull/11971#discussion_r227640632
cc: @eric-wieser

### CODE REVIEW

File: numpy/core/tests/test_shape_base.py [hunk 1]
```diff
@@ -375,39 +376,30 @@ def test_stack():
...
-    arrays, list_ndim, result_ndim, _ = _block_setup(arrays)
-    return _block_slicing(arrays, list_ndim, result_ndim)
-
-
 class TestBlock(object):
-    params = [dict(block=block),
-              dict(block=_block_force_concatenate),
-              dict(block=_block_force_slicing)]
+    @pytest.fixture(params=['block', 'force_concatenate', 'force_slicing'])
+    def block(request): # maybe needs self, not sure
```
Comment (eric-wieser) [inline numpy/core/tests/test_shape_base.py]: Assuming this works, should strip this comment out

Comment (hmaarrfk) [reply numpy/core/tests/test_shape_base.py]: man, what a mistake!

**Comment by hmaarrfk** [comment]
@eric-wieser I finally got around to fixing this. I'm glad I was explicit about my tests since it was always falling down to the default case. We needed the `self` parameter to the function

**Review by eric-wieser — APPROVED** [review]

File: numpy/core/tests/test_shape_base.py [hunk 2]
```diff
@@ -391,39 +392,32 @@ def test_stack():
...
+        def _block_force_slicing(arrays):
+            arrays, list_ndim, result_ndim, _ = _block_setup(arrays)
+            return _block_slicing(arrays, list_ndim, result_ndim)
+
+        if request.param == 'force_concatenate':
+            return _block_force_concatenate
+        elif request.param == 'force_slicing':
+            return _block_force_slicing
+        elif request.param == 'block':
+            return block
```
Comment (eric-wieser) [inline numpy/core/tests/test_shape_base.py]: Note you could also achieve this via the `ids` parameter to `fixture`:
```
@pytest.fixture(
    params=[block, _block_force_concatenate, _block_force_slicing],
    ids=['block', 'force_concatenate', 'force_slicing']
)
```
Which has the benefit of not needing the final `else`.
Not really a big deal either way.

Comment (hmaarrfk) [reply numpy/core/tests/test_shape_base.py]: that else statement is more a test on the test. It checks that I know how to use request (which I didn't).

**Comment by hmaarrfk** [comment]
Thanks for the review @eric-wieser

**Comment by charris** [comment]
Thanks @hmaarrfk .

**Pull request merged**